mainargs/src-3/Macros.scala

```diff
@@ -0,0 +1,152 @@
+package mainargs
+
+import scala.quoted._
+
+object Macros {
+  private def mainAnnotation(using Quotes) = quotes.reflect.TypeRepr.of[mainargs.main].typeSymbol
+  private def argAnnotation(using Quotes) = quotes.reflect.TypeRepr.of[mainargs.arg].typeSymbol
```

Comment:
This should be a bit more performant
```scala
  private def argAnnotation(using Quotes) = quotes.reflect.TypeRepr.typeConstructorOf[mainargs.arg].typeSymbol
```

Or maybe this one is even better
```scala
  private def argAnnotation(using Quotes) = quotes.reflect.Symbol.requiredClass("mainargs.arg")
```

Reply:
Same for `mainAnnotation`

Reply:
Done, thank you ✅